src/fuchur/scripts/electricity.py

```diff
@@ -0,0 +1,555 @@
+
+    sequences_df.index = building.timeindex(
+        year=str(temporal["scenario_year"]))
+
+    building.write_sequences(
+        "load_profile.csv",
+        sequences_df,
+        directory=os.path.join(datapackage_dir, "data/sequences"),
+    )
+
+
+def generation(config, datapackage_dir):
+    """
+    """
+    techmap = {
+        "ocgt": "dispatchable",
+        "ccgt": "dispatchable",
+        "pv": "volatile",
+        "wind_onshore": "volatile",
+        "wind_offshore": "volatile",
+        "biomass": "conversion",
+        "lithium_battery": "storage",
+        "acaes": "storage",
+    }
+
+    wacc = config["cost"]["wacc"]
+
+    technologies = pd.DataFrame(
+        Package(
+            "https://raw.githubusercontent.com/ZNES-datapackages/technology-cost/master/datapackage.json"
+        )
+        .get_resource("electricity")
+        .read(keyed=True)
+    )
+    technologies = (
+        technologies.groupby(["year", "tech", "carrier"])
+        .apply(lambda x: dict(zip(x.parameter, x.value)))
+        .reset_index("carrier")
+        .apply(lambda x: dict({"carrier": x.carrier}, **x[0]), axis=1)
+    )
+    technologies = technologies.loc[config["temporal"]["scenario_year"]].to_dict()
+
+    potential = (
+        Package(
+            "https://raw.githubusercontent.com/ZNES-datapackages/technology-potential/master/datapackage.json"
+        )
+        .get_resource("renewable")
+        .read(keyed=True)
+    )
+    potential = pd.DataFrame(potential).set_index(["country", "tech"])
+    potential = potential.loc[
+        potential["source"] == config["potential"]["renewables"]
+    ].to_dict()
+
+    for tech in technologies:
+        technologies[tech]["capacity_cost"] = technologies[tech][
+            "capacity_cost"
+        ] * config["cost"]["factor"].get(tech, 1)
+        if "storage_capacity_cost" in technologies[tech]:
+            technologies[tech]["storage_capacity_cost"] = technologies[tech][
+                "storage_capacity_cost"
+            ] * config["cost"]["factor"].get(tech, 1)
+
+    carrier = pd.read_csv(
+        os.path.join(fuchur.__RAW_DATA_PATH__,
+                     "carrier.csv"), index_col=[0, 1]
+    ).loc[("base", config["temporal"]["scenario_year"])]
+    carrier.set_index("carrier", inplace=True)
+
+    elements = {}
+
+    for r in config["buses"]["electricity"]:
+        for tech, data in technologies.items():
+            if tech in config["technologies"]["investment"]:
+
+                element = data.copy()
+                elements[r + "-" + tech] = element
+
+                if techmap.get(tech) == "dispatchable":
+                    element.update(
+                        {
+                            "capacity_cost": annuity(
+                                float(data["capacity_cost"]),
+                                float(data["lifetime"]),
+                                wacc,
+                            )
+                            * 1000,  # €/kW -> €/MW
```

Comment:
Best comment ever. Seriously. It's short and tells everybody something non-obvious they need to know. 👍